<launch>
  <arg name="spot" value="002"/>
  <arg name="map_type" default="lidar"/>
  <arg name="exp_name" value="spot_exp1_local_lab121"/>
  <arg name="local_search" default="false"/>

  <!-- If we are doing local search, then the search region center is published
       at /local_region_center; Otherwise, it is spot's hand pose -->
  <group if="$(arg local_search)">

    <!-- publish region center -->
    <node name="local_search_center_publisher"
          pkg="genmos_object_search_ros"
          type="lab121_local_$(arg map_type)_search_region_center.py" output="screen"></node>

    <!-- Run GENMOS object search decision making interface (POMDP planner) -->
    <node name="genmos_ros" ns="spot_$(arg spot)"
          pkg="genmos_object_search_ros" type="run_search.py" output="screen">
      <!-- So strange. you can't do ~config -->
      <rosparam param="config" command="load"
                file="$(find genmos_object_search_ros)/config/experiments/$(arg exp_name).yaml"/>
      <param name="robot_id" value="spot_$(arg spot)"/>
      <param name="world_frame" value="graphnav_map"/>
      <param from="dynamic_update" to="False"/>
      <remap from="~search_region_center" to="/local_region_center" />
      <remap from="~search_region_cloud_3d" to="/local_cloud_publisher/region_points"/>
      <remap from="~search_region_cloud_2d" to="/local_cloud_publisher/region_points"/>
      <remap from="~robot_pose" to="/spot_hand_pose"/>
      <remap from="~object_detections" to="/spot/segmentation/hand/result_boxes3d"/>
      <remap from="~vision_info" to="/spot/segmentation/hand/vision_info"/>
      <remap from="~action_done" to="/spot_$(arg spot)/action_executor/done"/>
    </node>
  </group>

  <group unless="$(arg local_search)">

    <!-- Run GENMOS object search decision making interface (POMDP planner) -->
    <node name="genmos_ros" ns="spot_$(arg spot)"
          pkg="genmos_object_search_ros" type="run_search.py" output="screen">
      <!-- So strange. you can't do ~config -->
      <rosparam param="config" command="load"
                file="$(find genmos_object_search_ros)/config/experiments/$(arg exp_name).yaml"/>
      <param name="robot_id" value="spot_$(arg spot)"/>
      <param name="world_frame" value="graphnav_map"/>
      <remap from="~search_region_center" to="/spot_hand_pose" />
      <remap from="~search_region_cloud_3d" to="/local_cloud_publisher/region_points"/>
      <remap from="~search_region_cloud_2d" to="/local_cloud_publisher/region_points"/>
      <remap from="~robot_pose" to="/spot_hand_pose"/>
      <remap from="~object_detections" to="/spot/segmentation/hand/result_boxes3d"/>
      <remap from="~vision_info" to="/spot/segmentation/hand/vision_info"/>
      <remap from="~action_done" to="/spot_$(arg spot)/action_executor/done"/>
    </node>

  </group>


</launch>
